src/main/java/edu/harvard/iq/dataverse/authorization/providers/builtin/DataverseUserPage.java

```diff
@@ -360,8 +359,12 @@ public String save() {
 
 
             return redirectPage + (!redirectPage.contains("?") ? "?" : "&") + "faces-redirect=true";            
-            
-        } else {
+
+        //Happens if user is logged out while editing
+        } else if (!session.getUser().isAuthenticated()) {
```

Comment:
Password Change mode is used to change the password. The bug technically still exists there, but it requires knowing the old password anyway, so it's kind of irreverent.